src/main/java/frc/robot/PowerController.java

```diff
@@ -0,0 +1,168 @@
+import frc.robot.subsystems.tank.TankSubsystem;
+import static frc.robot.Constants.TankConstants.*;
+
+public class PowerController {
+
+    private static PowerDistribution PDP = new PowerDistribution();
+
+    private ControllableSubsystem tankSubsystem;
+
+//    private ControllableSubsystem climbSubsystem;
+
+   // private ControllableSubsystem intakeSubsystem;
+
+   // private ControllableSubsystem shooterSubsystem;
+
+    private int currentCurrentLimit;
+
+    private int totalSustainableCurrent;
+
+    //TODO find current max value
+
+
+    private List<ControllableSubsystem> subsystems;
+
+    public PowerController(ControllableSubsystem tankSubsystem) {
+        subsystems = new ArrayList<ControllableSubsystem>();
+
+        subsystems.add(tankSubsystem);
+
+        this.tankSubsystem = tankSubsystem;
+
+        currentCurrentLimit = 350;
+        //should be same as limit in tank subsystem
+
+        //TODO Calculate total sustainable current
+        totalSustainableCurrent = 200;
+
+    }
+
+    public void check() {
+
+        System.out.println("Checking for a brownout...");
+        // Check PDP voltage; if close to a brownout:
+        if (PDP.getVoltage() < 7) {
+            System.out.println("close to brownout!");
+            setBrownoutScaling();
+        }
+
+        // Sum up total current drawn from all subsystems
+        int totalCurrent = (int)PDP.getTotalCurrent();
+        int nonTankCurrent = totalCurrent - getCurrentDrawnFromPDP(fLeftMotorPort,fRightMotorPort,bLeftMotorPort,bRightMotorPort);
+        System.out.println("total current drawn: " + totalCurrent + 
+                            "; non tank current drawn: " + nonTankCurrent);
+
+        //if current goes over sustainable current...
+        // Set scale of the current drawn by TankSubsystem
+        if (totalCurrent > totalSustainableCurrent) {
+
+            scale(totalCurrent, null);
+
+            /* // New scale = current available after other components draw divided by current
+            // drawn by tank
+            tankSubsystem.setCurrentLimit((totalSustainableCurrent - nonTankCurrent));
+            System.out.println("New limit for tank subsystem: " + currentCurrentLimit);
+            */
+        }
+
+      
+    }
+
+    public static int getCurrentDrawnFromPDP(int... PDPChannel) {
+        int sum = 0;
+        int channels = 0;
+
+        for (int channel : PDPChannel) {
+            sum += PDP.getCurrent(channel);
+            channels++;
+        }
+
+        if ((channels == 16) && (sum != PDP.getTotalCurrent())) {
+            System.out.println("something is wrong, total current does not match");
+        }
+
+        return sum;
+    }
+
+
+    private void setBrownoutScaling() {
+
+        //scale all subsystems by set (sensible) amount
+        for (ControllableSubsystem subsystem : subsystems) {
+            int currDrawn = subsystem.getTotalCurrentDrawn();
+            int tempscale = currDrawn - (int)(currDrawn*0.2);
```

Comment:
I'm concerned about all the integer arithmetic. Do we know what the expected range of values are here?

Reply:
It's hard, because `PDP.getCurrent()` returns `double`, but `CANSparkMax` expects an `int` for `setSmartCurrentLimit()`. I can get rid of some of the intermediary `int` casts but it will still have to be cast to `int` in the end for scaling.

Reply:
Yes, let's do that. Since you measure in `double` let's use that for all the math and then cast (or better yet, design a rounding function) to an int for the current limit. CS math isn't quite the same as pencil-and-paper math, so it's best to stick with `double` for the math (that only has issues with equals, not multiplication and division).